dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04205MigrateVanityURLToContent.java

```diff
@@ -0,0 +1,249 @@
+								+ title + ", URL: " + oldURL + ", URI: " + oldUri);
+				dc.setSQL(DELETE_INODE_BY_INODE_QUERY);
+				dc.addParam(inode);
+				dc.loadResult();
+			} else {
+
+				String uuid = UUIDGenerator.generateUuid();
+				
+				int lang = Integer.parseInt(findHostLang(site)); //Gets the lang of the Site, so when saving the Vanity URL it's the same lang
+
+				// Insert on Identifier Table
+				dc.setSQL(INSERT_IDENTIFIER_QUERY);
+				dc.addParam(uuid); // identifier
+				dc.addParam("/"); // parent_path
+				dc.addParam("content." + inode); // asset_name
+				dc.addParam(Host.SYSTEM_HOST); // host_inode
+				dc.addParam("contentlet"); // asset_type
+				dc.loadResult();
+
+				// Insert on Contentlet Table
+		        if (DbConnectionFactory.isMySql()) {
+		            // Use correct escape char when using reserved words as column names
+		        	INSERT_CONTENTLET_QUERY = INSERT_CONTENTLET_QUERY.replaceAll("\"", "`");
+		        }
+				dc.setSQL(INSERT_CONTENTLET_QUERY);
+				dc.addParam(inode); // inode
+				dc.addParam(false); // show_on_menu
+				dc.addParam(0); // sort_order
+				dc.addParam(title); // title
+				dc.addParam(new Date()); // mod_date
+				dc.addParam("dotcms.org.1"); // mod_user
+				dc.addParam(title); // friendly_name
+				dc.addParam("8e850645-bb92-4fda-a765-e67063a59be0"); // structure_inode
+				dc.addParam(uuid); // identifier
+				dc.addParam(lang); // language_id
+				dc.addParam(title); // text1(title)
+				dc.addParam(site); // text2(site)
+				dc.addParam(uri); // text3(uri)
+				dc.addParam(oldUri); // text4(forwardTo)
+				dc.addParam(301); // integer1(action)
+				dc.addParam(0); // integer2(sort)
+				for (int i = 0; i < 48; i++) {
+					dc.addParam(0); // all other integer and float fields
+				}
+				for (int i = 0; i < 25; i++) {
+					dc.addParam(false); // all bool fields
+				}
+
+				dc.loadResult();
+
+				// Insert on Contentlet_Version_Info
+				dc.setSQL(INSERT_CONTENTLET_VERSION_INFO_QUERY);
+				dc.addParam(uuid); // identifier
+				dc.addParam(lang); // lang
+				dc.addParam(inode); // live_inode
+				dc.addParam(inode); // working_inode
+				dc.addParam(false); // deleted
+				dc.addParam(new Date()); // locked_on
+				dc.addParam(new Date()); // version_ts
+				dc.loadResult();
+			}
+		}
+
+	}
+	
+	/**
+	 * This method splits the old Url into the site and uri. 
+	 * Also make the call for the {@link #findHostId(String)} that returns the host identifier
+	 * 
+	 * @param url. The value of the url column on the virtual_link table.
+	 * @return a map with the host identifier and uri
+	 * @throws DotDataException
+	 */
+	private Map<String,String> splitURL(String url) throws DotDataException{
+		String hostId = Host.SYSTEM_HOST;
+		
+		int hostSplit = url.indexOf(":");
+		if(hostSplit != -1){
+			String hostName = url.substring(0, url.lastIndexOf(":"));
+			url = url.substring(hostSplit+2);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "url". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)